src/config.c

```diff
@@ -607,60 +619,268 @@ int parseCommandLine(int argc,char **argv,ParodusCfg * cfg)
+
+		if(r_count == 1)
+		{
+			ParodusInfo("Curl Ip resolve option set as V4 mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V4);
+		}
+		else if(r_count == 2)
+		{
+			ParodusInfo("Curl Ip resolve option set as V6 mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_V6);
+		}
+		else
+		{
+			ParodusInfo("Curl Ip resolve option set as default mode\n");
+			curl_easy_setopt(curl, CURLOPT_IPRESOLVE, CURL_IPRESOLVE_WHATEVER);
+		}
+
+		/* set the cert for client authentication */
+		curl_easy_setopt(curl, CURLOPT_SSLCERT, get_parodus_cfg()->client_cert_path);
+
+		curl_easy_setopt(curl, CURLOPT_CAINFO, get_parodus_cfg()->cert_path);
+
+		/* disconnect if it is failed to validate server's cert */
+		curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 1L);
+
+		/* Perform the request, res will get the return code */
+		res = curl_easy_perform(curl);
+		ParodusInfo("themis curl response code is %d\n", res);
+
+		time_res = curl_easy_getinfo(curl, CURLINFO_TOTAL_TIME, &total);
+		if(time_res == 0)
+		{
+			ParodusInfo("curl response Time: %.1f seconds\n", total);
+		}
+		if(transaction_uuid !=NULL)
+		{
+			free(transaction_uuid);
+			transaction_uuid = NULL;
+		}
+		if(mac_header !=NULL)
+		{
+			free(mac_header);
+			mac_header = NULL;
+		}
+		if(serial_header !=NULL)
+		{
+			free(serial_header);
+			serial_header = NULL;
+		}
+		if(uuid_header !=NULL)
+		{
+			free(uuid_header);
+			uuid_header = NULL;
+		}
+		curl_slist_free_all(headers_list);
+		if(data.data)
+		{
+			strncpy(newToken, data.data, len);
+			free(data.data);
+			data.data = NULL;
+		}
+		if(res != 0)
+		{
+			ParodusError("curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
+			curl_easy_cleanup(curl);
+			return -1;
+		}
+		else
+		{
+			ParodusInfo("cURL success\n");
+		}
+		curl_easy_cleanup(curl);
+	}
+	else
 	{
-		ParodusError("Failed to create new token\n");
+		ParodusError("curl init failure\n");
+		return -1;
 	}
+
+	return 0;
 }
 
 /*
-* Fetches authorization token from the output of read script. If read script returns "ERROR"
-* it will call createNewAuthToken to create and read new token 
+* @brief Fetches authorization token and update to parodus config.
+  This will do curl retry in case of any failure till it reaches max curl retry count.
+ * @param[in] cfg Global parodus config structure to update webpa_auth_token
 */
 
 void getAuthToken(ParodusCfg *cfg)
 {
-	//local var to update cfg->webpa_auth_token only in success case
-	char output[4069] = {'\0'} ;
-	
-	if( strlen(cfg->token_read_script) !=0 && strlen(cfg->token_acquisition_script) !=0)
-    	{
-		execute_token_script(output,cfg->token_read_script,sizeof(output),cfg->hw_mac,cfg->hw_serial_number);
-		
-	    	if ((strlen(output) == 0))
-	    	{
-			ParodusError("Unable to get auth token\n");
-		}
-		else if(strcmp(output,"ERROR")==0)
+	int status = -1;
+	int retry_count = 0;
+
+	if( cfg->client_cert_path !=NULL && strlen(cfg->client_cert_path) !=0 )
```

Comment:
Add check such that curl request is not attempted if mac is NULL or client cert is not available and log this case

Reply:
incorporated.